file/src/main/java/org/apache/zeppelin/file/HDFSFileInterpreter.java

```diff
@@ -0,0 +1,225 @@
+    public int blockSize;
+    public int childrenNum;
+    public int fileId;
+    public String group;
+    public long length;
+    public long modificationTime;
+    public String owner;
+    public String pathSuffix;
+    public String permission;
+    public int replication;
+    public int storagePolicy;
+    public String type;
+    public String toString() {
+      String str = "";
+      str += "\nAccessTime = " + accessTime;
+      str += "\nBlockSize = " + blockSize;
+      str += "\nChildrenNum = " + childrenNum;
+      str += "\nFileId = " + fileId;
+      str += "\nGroup = " + group;
+      str += "\nLength = " + length;
+      str += "\nModificationTime = " + modificationTime;
+      str += "\nOwner = " + owner;
+      str += "\nPathSuffix = " + pathSuffix;
+      str += "\nPermission = " + permission;
+      str += "\nReplication = " + replication;
+      str += "\nStoragePolicy = " + storagePolicy;
+      str += "\nType = " + type;
+      return str;
+    }
+  }
+
+  /**
+   * Status of one file
+   *
+   * matches returned JSON
+   */
+  public class SingleFileStatus {
+    public OneFileStatus FileStatus;
+  }
+
+  /**
+   * Status of all files in a directory
+   *
+   * matches returned JSON
+   */
+  public class MultiFileStatus {
+    public OneFileStatus[] FileStatus;
+  }
+
+  /**
+   * Status of all files in a directory
+   *
+   * matches returned JSON
+   */
+  public class AllFileStatus {
+    public MultiFileStatus FileStatuses;
+  }
+
+  // tests whether we're able to connect to HDFS
+
+  private void testConnection() {
+    try {
+      if (isDirectory("/"))
+        logger.info("Successfully created WebHDFS connection");
+    } catch (Exception e) {
+      logger.error("testConnection: Cannot open WebHDFS connection. Bad URL: " + "/", e);
+      exceptionOnConnect = e;
+    }
+  }
+
+  @Override
+  public void open() {
+    testConnection();
+  }
+
+  @Override
+  public void close() {
+  }
+
+  private String listDir(String path) throws Exception {
+    return cmd.runCommand(cmd.listStatus, path, null);
+  }
+
+  private String listPermission(OneFileStatus fs){
+    String s = "";
+    s += fs.type.equalsIgnoreCase("Directory") ? 'd' : '-';
+    int p = Integer.parseInt(fs.permission, 16);
+    s += ((p & 0x400) == 0) ? '-' : 'r';
+    s += ((p & 0x200) == 0) ? '-' : 'w';
+    s += ((p & 0x100) == 0) ? '-' : 'x';
+    s += ((p & 0x40)  == 0) ? '-' : 'r';
+    s += ((p & 0x20)  == 0) ? '-' : 'w';
+    s += ((p & 0x10)  == 0) ? '-' : 'x';
+    s += ((p & 0x4)   == 0) ? '-' : 'r';
+    s += ((p & 0x2)   == 0) ? '-' : 'w';
+    s += ((p & 0x1)   == 0) ? '-' : 'x';
+    return s;
+  }
+  private String listDate(OneFileStatus fs) {
+    return new SimpleDateFormat("yyyy-MM-dd HH:mm").format(new Date(fs.modificationTime));
+  }
+  private String ListOne(String path, OneFileStatus fs) {
+    if (args.flags.contains(new Character('l'))) {
+      String s = "";
+      s += listPermission(fs) + "\t ";
```

Comment:
nit: probably should use http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html to avoid creating all these intermediate strings